core/src/main/java/org/mskcc/oncotree/utils/CacheUtil.java

```diff
@@ -48,14 +48,23 @@ public static void resetTumorTypesByVersion(Version version) {
         }
         if (tumorTypes.containsKey(version)) {
             logger.debug("getTumorTypesByVersion() -- found '" + version.getVersion() + "' in cache");
-            return tumorTypes.get(version);
+            return getUnmodifiableTumorTypesByVersion(tumorTypes.get(version));
         } else {
             logger.debug("getTumorTypesByVersion() -- did NOT find '" + version.getVersion() + "' in cache, getting now");
             tumorTypes.put(version, TumorTypesUtil.getTumorTypesByVersionFromRaw(version));
-            return tumorTypes.get(version);
+            return getUnmodifiableTumorTypesByVersion(tumorTypes.get(version));
         }
     }
 
+    private static Map<String, TumorType> getUnmodifiableTumorTypesByVersion(Map<String, TumorType> tumorTypeMap) {
```

Comment:
In some ways .. since we are passing back a full, detached clone of the data it is less important that the returned map be unmodifiable. We are safe now.